src/adapter/src/catalog/storage.rs

```diff
@@ -358,6 +371,31 @@ async fn migrate(
             txn.cluster_replicas.update(|_k, v| Some(v.clone()))?;
             Ok(())
         },
+        // Attributes were added to role definitions.
+        //
+        // Introduced in v0.45.0
+        |txn: &mut Transaction<'_>, _now, _bootstrap_args| {
+            for (role_key, role_value) in txn.roles_old.delete(|_, _| true) {
+                let is_materialize_role = role_value.name == MATERIALIZE_ROLE;
+                let is_mz_introspection_role = role_value.name == INTROSPECTION_USER.name;
+                let migrated_role_value = RoleValue {
+                    role: SerializedRole {
+                        name: role_value.name,
+                        attributes: RoleAttributes {
+                            super_user: !is_mz_introspection_role,
+                            inherit: true,
+                            create_role: is_materialize_role,
+                            create_db: is_materialize_role,
+                            create_cluster: is_materialize_role,
+                            create_persist: is_materialize_role,
+                            can_login: true,
+                        },
+                    },
+                };
+                txn.roles.insert(role_key, migrated_role_value)?;
```

Comment:
@mjibson Do you know if there's a better way to do this? What I'd really like to do is:
1. Get all roles out of the collection named "role" using the old role value format.
2. Delete all roles out of the collection name "role".
3. Insert some roles into the collection named "role" using the new role value format.

It seems like that should be possible but maybe we don't have the right API to do this. Instead I think I would have to spread this out over 4 releases:
release 1: Delete all values out of "roles" and insert values into "roles_new".
release 2: Delete collection "roles",
release 3: Create new collection "roles", delete all values out of "roles_new" and insert values into "roles".
release 4: Delete collection "roles_new".

Reply:
I guess there's no way to do step 1 safely, because we don't know if it's a fresh installation in which case "roles" would be populated with the new value type or it's an old installation in which case "roles" would be populated with the new value type.

Reply:
Will I ever be able to remove the `Option` or is that stuck there forever?

Reply:
I think I should do some serde reading. Maybe it's just smart enough to fill in default values for new bool fields. Then I can just update all of them.

Reply:
Ok, I think I really cleaned this up using default values during deserialization. Let me know what you think.